File: 739-daily-temperatures/daily-temperatures.java
class Solution {
    class Pair{
        int val;
        int index;
        Pair(int val, int index){
            this.val = val;
            this.index = index;
        }
    }
    public int[] dailyTemperatures(int[] temperatures) {
            int i = 0;  
            int[] result = new int[temperatures.length];
            Arrays.fill(result, 0);
            Stack<Pair> stack = new Stack<>();
            while(i < temperatures.length){
                while(!stack.isEmpty() && stack.peek().val < temperatures[i]){
                    Pair curr = stack.pop();
                    result[curr.index] = i - curr.index;
                }
                Pair add = new Pair(temperatures[i], i);
                stack.push(add);
                i++;
            }
            return result;
    }   
}

File: 739-daily-temperatures/daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = []
        res = [0] * (len(temperatures))
        for i in range(0, len(temperatures)):
            while stack and temperatures[i] > stack[-1][1]:
                curr = stack.pop()
                res[curr[0]] = i - curr[0]
            stack.append((i, temperatures[i]))
        return res

        